provider/core/src/erased.rs

```diff
@@ -66,10 +66,13 @@ impl ZeroCopyFrom<dyn ErasedDataStruct> for &'static dyn ErasedDataStruct {
 pub struct ErasedDataStructMarker {}
 
 impl DataMarker<'static> for ErasedDataStructMarker {
-    type Yokeable = &'static dyn ErasedDataStruct;
-    type Cart = dyn ErasedDataStruct;
+    type Yokeable = ErasedDataYokeable;
```

Comment:
Question: does it work to make it a straight `Box<dyn ErasedDataStruct>`?  Do you need the wrapper type at all?

Reply:
Yes, it's a privacy encapsulation so that others cannot construct e.g. a `RcBuf` variant `Box<dyn EDS>` and then cause an error.

Reply:
But it would work to remove the indirection; I feel this is cleaner.